### ISSUE TITLE
Make builds for MacOS on ARMv8 M1 chip

### ISSUE THREAD

**Issue by XVilka** [rizinorg/cutter#2504]
It is very important to make Cutter available for the MacOS on modern Apple hardware - M1-based laptops.

**Comment by ReedEggleston** [comment]
Howdy! Is there a list of things that need to be done for a successful M1 build? Unsure if this would warrant a separate issue thread; neither x64 build or brew cask installation seem work currently.

**Comment by xarkes** [comment]
Hello, I think there is no official build of `Qt` for Apple M1 yet, which is probably the only real blocking point for Cutter.
When you say "x64 build seem [not to] work currently" do you mean that Cutter through Rosetta does not work?

**Comment by ReedEggleston** [comment]
> When you say "x64 build seem [not to] work currently" do you mean that Cutter through Rosetta does not work?
That is correct; it fails to start.

**Comment by karliss** [comment]
Last time I checked Cutter through Rosetta on M1 worked quite well (except debugging), both with prebuilt x64 binaries and when building locally. Although at some point homebrew changed the way it works through Rosetta causing some issues for local builds, but that might have been me forgetting to use correct terminal window with Rosetta enabled.

**Comment by karliss** [comment]
Just tried the latest CI build on M1 mac mini and it worked. There was the usual warning about developer Identity, but that's to be expected.

Yes biggest blocker is probably Qt. Last time time checked it was also a blocker for native homebrew on M1. Supposedly native homebrew on M1 is now available, not sure if it means that they resolved the Qt building issues or it's not available for now. This would be a good moment to recheck the current status of it.  In case the M1 Qt changes are only included in Qt6 it would require finishing the Qt6  support for Cutter. 

Other potential changes:
* rizin compilation, does it work now?  What about fat binary builds? I remember there were some discussions or issue for that, due to meson having less stuff builtin compared to CMake. 
* fat binary builds and deciding if we want to publish two separate packages or single binary for both architectures. From the CMake side things are simple it supports it quite well, it's just the matter of passing single flag. But with the additional external libraries like rizin and anything indirectly required by Qt and python  and copying the into app bundle to make it selfcontained it might be a bit more tricky. Might also need some architecture handling within bundled rizin script to ensue that selected set of architectures matches current cmake options.
* building cutter-deps for M1, either separately or as fat binary builds depending on choices above.

**Comment by XVilka** [comment]
To track the Qt porting progress see this bug: https://bugreports.qt.io/browse/QTBUG-85279

**Comment by ReedEggleston** [comment]
> Just tried the latest CI build on M1 mac mini and it worked. There was the usual warning about developer Identity, but that's to be expected.

I was able to get the standalone build to work; I had to install x86_64 gettext via homebrew.

**Comment by thestr4ng3r** [comment]
Not sure what Qt still have on their TODO, but a completely native arm64 build with Qt 5.15.2 from macports (source install) works just fine.
For distributing builds I think we will want to provide separate packages for each architecture instead of fat binaries since the code size with Qt and everything included is quite large.

**Comment by karliss** [comment]
Size argument needs to be evaluated more carefully. But due to issues with packaging all direct and indirect dependencies it might still be easier to start with split builds.

Current Cutter.pkg is 53.4 MB, but after extracting the app it becomes 497MB. From which
* 375 MB are Ghidra sleigh files
* 50 MB python libs (.py)
It gives an upper bound of 72MB of binaries and 15% increase for Cutter.app. It's hard to predict what what will happen with .dmg size.

**Comment by timtianyang** [comment]
For whoever landed here, this is what I use to install cutter on ARM Macs:
`
# install cutter, assuming arm homebrew is installed.
brew install --cask cutter 

# install x86 homebrew 
arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"

# install x86 gettext
arch -x86_64 brew install gettext
`

**Comment by n0thing2speak** [comment]
https://github.com/crystalidea/macdeployqt-universal

**Comment by n0thing2speak** [comment]
> It looks like the QT ticket has been closed in 6.2. Just built it on my m1 mac and it is super smooth running natively without rosetta. Do we consider adding a ARM release image as well? It's nice to have decompilers packaged together!

Can you release your build version？

**Comment by MarkusZoppelt** [comment]
All the dependencies are ready to generate cutter builds for macOS arm64:

Install all the dependencies with brew:

    brew install cmake qt5 meson ninja

Notice that homebrew for arm64 puts qt5 here: `/opt/homebrew/opt/qt5` so adapt your build steps as such:


```
mkdir build
cd build
cmake .. -DCMAKE_PREFIX_PATH=/opt/homebrew/opt/qt5
make
```

**Comment by thestr4ng3r** [comment]
Official macOS/arm64 builds are available since 2.1.0 and will be automated in CI soon.
